middleware/proxy/upstream.go

```diff
@@ -150,6 +153,14 @@ func RegisterPolicy(name string, policy func() Policy) {
 	supportedPolicies[name] = policy
 }
 
+// AddProxyHeader adds a proxy header.
+func addProxyHeader(header, value string) {
+	if proxyHeaders == nil {
+		proxyHeaders = make(map[string][]string)
```

Comment:
Can we spare ourselves this grief just be declaring proxyHeaders above and making it at the same time? i.e. `var proxyHeaders = make(http.Header)`